app/views/projects/_form.html.haml

```diff
@@ -24,15 +24,24 @@
         = text_field_tag field_name, val, :disabled => @readonly
         %br
       - unless @readonly
-        = link_to 'Add new', '', :class => 'add'
-        .field.new
-          = text_field_tag "project[configs_attributes][#{index}][new][]", '', :id => nil, :class => 'newf'
-          \ :
-          = text_field_tag "project[configs_attributes][#{index}][new][]", '', :id => nil, :class => 'newf'
-          %br
+        - if @project_metrics[name].respond_to?(:credentials)
```

Comment:
it looks like you were moving this logic into the controller, but then it wasn't removed from here